src/dune/install_rules.mli

```diff
@@ -3,6 +3,16 @@ open Stdune
 val gen_rules :
   Super_context.t -> dir:Path.Build.t -> Build_system.Subdir_set.t
 
-val init_meta_and_dune_package : Super_context.t -> dir:Path.Build.t -> unit
+(** Generate rules for [.dune-package] and [META.<package-name>] files in a
+    given super context and directory. Surprisingly, the current implementation
+    generates rules for {b all} directories of the project, which are filtered
+    out later in the [build_system.ml]. This function is called multiple times
+    for the same project (with different [dir]), so we memoize it to avoid
+    duplicating work. *)
+val meta_and_dune_package_rules : Super_context.t -> dir:Path.Build.t -> unit
+
+(* TODO: A seemingly more sensible approach for [meta_and_dune_package_rules]
+   is to only generate rules for the given directory, or stop taking [dir]
+   altogether and take [project] instead. *)
```

Comment:
The latter approach seems to be worth trying. Wouldn't it be enough to just generate rules for the project's root dir to make it work?

Reply:
Things can be cleaned up around here for sure. 
I think my preferred way would be to collect `meta_and_dune_package_rules_impl` and `install_rules` together into a single `Scheme.t` and leave the per-directory projection to `Scheme.evaluate`.

For this PR, cleanup is not the focus, so I think we should merge it.

Reply:
Added to comment:

```
aalekseyev: actually I think we should just remove
   [meta_and_dune_package_rules] from the interface and have [gen_rules]
   do everything.
```